### PR TITLE
Remove TAPs from CT-RAMP

### PR DESCRIPTION
**Pull request by i-am-sijia**
This PR works on the Java CT-RAMP changes as part of the [TM2.2.1 design and work plan](https://app.box.com/file/1116904392850?s=j5g911ye9cizq2kgxhyjp8lsla5zkk9e). The Java code changes touches on:

- [x] Removing TAP data manager
- [x] Removing TAPs in the best transit path calculator, replacing them with TAZ level best path
- [x] Removing TAPs in the logsum calculation
- [x] Removing TAPs in mode choice
- [x] Removing TAPs in trip records
- [x] Commenting out PNR model, it will be replaced by the PNR simulation in Emme network assignment

All relevant UECs are also updated to:

- [x] Use TAZ transit skims
- [x] Refer to non-TAP Java getter methods

The team will test run CT-RAMP with the changes before merging this PR.

### CODE REVIEW

**Comment by lmz** [comment]
Hi @i-am-sijia - would you mind adding some more documentation to this pull request?  What does this mean to the user?

**Comment by DavidOry** [comment]
A key point is that this version of the TM2 CT-RAMP code will not be backwards compatible with the TAM or any other releases that are in use.

File: core/src/java/com/pb/mtctm2/abm/accessibilities/BestTransitPathCalculator.java [hunk 1]
```diff
@@ -736,9 +351,9 @@ public float calcPathUtility(TransitWalkAccessDMU walkDmu, TransitDriveAccessDMU
      * @param departPeriod Departure time period - 1 = AM period, 2 = PM period, 3 =OffPeak period
      * @param debug boolean flag to indicate if debugging reports should be logged
      * @param logger Logger to which debugging reports should be logged if debug is true
-     * @return double[][] Array of best tap pair values - rows are N-path, columns are orig tap, dest tap, skim set, utility
+     * @return double transit utility
      */
-    public double[][] getBestTapPairs(TransitWalkAccessDMU walkDmu, TransitDriveAccessDMU driveDmu, int accMode, int origMgra, int destMgra, int departPeriod, boolean debug, Logger myLogger, float odDistance)
+    public double calcPersonSpecificUtilities(int pTaz, int aTaz, TransitWalkAccessDMU walkDmu, TransitDriveAccessDMU driveDmu, int accMode, int origMgra, int destMgra, int departPeriod, boolean debug, Logger myLogger, float odDistance)
```
Comment (DavidOry) [inline core/src/java/com/pb/mtctm2/abm/accessibilities/BestTransitPathCalculator.java:356]: @lmz, @AshishKuls, @i-am-sijia 

Per our call, a better approach would be to:

1. Start with a walk and drive specific versions of the `calcPathUtility` method. The methods would have the same name, but one would expect a walkDMU and the other a driveDMU.
2. Propagate this pattern up to this method, again creating two versions, one for walk and one for drive.  

When invoking these methods when accessing walk paths, the code would therefore not need the drive details. 

Given our current focus on getting TM2.2.1 up and running and the likelihood that making this change will likely break something else, we will note this for now but move forward.

**Comment by DavidOry** [comment]
Adding two debug prints for the BestTransitPath calculator. We found that `tm2py` was already updating the number of transfers for the PnR dummies -- see [here](https://github.com/BayAreaMetro/tm2py/blob/bart_transit_improvements/tm2py/components/network/transit/transit_skim.py#L374). @arashasadabadi: please commit the updated UEC with the correction here. The other utility calculations specific to the PnR dummy buses were working as expected.
[2158682_pnr.txt](https://github.com/BayAreaMetro/travel-model-two/files/13756254/2158682_pnr.txt)
[2142759_pnr.txt](https://github.com/BayAreaMetro/travel-model-two/files/13756255/2142759_pnr.txt)

**Pull request merged**